wordvecspace/disk.py

```diff
@@ -182,3 +191,115 @@ def get_word_occurrences(self, words_or_indices, raise_exc=False):
+        [[ 1.5464  0.4876  0.3017 ...,  1.2492  1.2451  0.8925]
+         [ 1.0436  0.9995  1.0913 ...,  0.6996  0.8014  1.1608]]
+        >>> print(wv.get_distances(["andhra"]))
+        [[ 1.5418  0.7153  0.277  ...,  1.1657  1.0774  0.7036]]
+        >>> print(wv.get_distances(["andhra"], metric='euclidean'))
+        [[ 1.756   1.1961  0.7443 ...,  1.5269  1.4679  1.1862]]
+        '''
+        r = row_words_or_indices
+        c = col_words_or_indices
+
+        if not metric:
+            metric = self.metric
+
+        if not isinstance(r, (tuple, list)):
+            r = [r]
+
+        if c:
+            if not isinstance(c, (tuple, list)):
+                c = [c]
+
+        if metric == 'angular':
+            row_vectors = self.get_word_vectors(r, normalized=True, raise_exc=raise_exc)
+
+            col_vectors = self._f.getmany(0, self.nvecs)
+            if c:
+                col_vectors = self.get_word_vectors(c, normalized=True, raise_exc=raise_exc)
+
+            if len(r) == 1:
+                nvecs, dim = col_vectors.shape
+
+                vec_out = self._make_array((len(col_vectors), len(row_vectors)), dtype=np.float32)
+                res = self._perform_sgemv(col_vectors, row_vectors, vec_out, nvecs, dim).T
+
+            else:
+                mat_out = self._make_array((len(row_vectors), len(col_vectors)), dtype=np.float32)
+                res = self._perform_sgemm(row_vectors, col_vectors, mat_out)
+
+            return 1 - res
+
+        elif metric == 'euclidean':
+            row_vectors = self.get_word_vectors(r, raise_exc=raise_exc)
+
+            if c:
+                col_vectors = self.get_word_vectors(c, raise_exc=raise_exc)
+            else:
+                col_vectors = self._f.getmany(0, self.nvecs)
+
+            return distance.cdist(row_vectors, col_vectors, 'euclidean')
+
+    DEFAULT_K = 512
+
+    def get_nearest(self, words_or_indices, k=DEFAULT_K, combination=False, metric=None, raise_exc=False):
+        '''
+        >>> wv = WordVecSpaceDisk(DATAFILE_ENV_VAR)
+        >>> print(wv.get_nearest("india", 20))
+        [509, 3389, 486, 523, 7125, 16619, 4491, 12191, 6866, 8776, 15232, 14208, 5998, 21916, 5226, 6322, 4343, 6212, 10172, 6186]
+        >>> print(wv.get_nearest("india", 20, metric='euclidean'))
+        [509, 3389, 486, 523, 7125, 16619, 4491, 12191, 6866, 8776, 15232, 14208, 5998, 21916, 5226, 6322, 4343, 6212, 10172, 6186]
+        '''
+
+        if not metric:
+            metric = self.metric
+
+        distances = self.get_distances(words_or_indices, metric=metric, raise_exc=raise_exc)
+
+        ner = []
+        for dist in distances:
+            dist = pd.Series(dist.reshape((len(dist,))))
+            dist = dist.nsmallest(k).keys()
+            ner.append(list(dist))
+
+        if combination:
+            return list(set(ner[0]).intersection(*ner))
+
+        else:
+            return ner if isinstance(words_or_indices, (list, tuple)) else ner[0]
```

Comment:
@prasannababuAddagiri here we no need to check the else condition. If the condition satisfies we can return there itself.
Ex:
```python
>>> def test(x):
...     if x == 1:
...             return x
...     return x
```